monai/handlers/roc_auc.py

```diff
@@ -96,6 +98,13 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
     def compute(self):
         _prediction_tensor = torch.cat(self._predictions, dim=0)
         _target_tensor = torch.cat(self._targets, dim=0)
+
+        if dist.is_available() and dist.is_initialized() and not self._is_reduced:
+            dist.barrier()
+            _prediction_tensor = dist.gather(_prediction_tensor)
```

Comment:
`dist.gather` -> `dist.all_gather`

Reply:
Thanks for correcting me, updated in the latest commit.